crates/forge_main/src/title_display.rs

```diff
@@ -41,12 +41,15 @@ impl TitleDisplay {
 
         // Add timestamp if requested
         if self.with_timestamp {
-            buf.push_str(
+            let timestamp_str = if let Some(timestamp) = self.inner.timestamp {
+                // Use replay timestamp if provided
+                let local_time: chrono::DateTime<Local> = timestamp.into();
+                format!("[{}] ", local_time.format("%H:%M:%S"))
+            } else {
+                // Use current time for live conversations
```

Comment:
display should only be concerned with formatting and not with generating timestamps.